contracts/src/v0.8/dev/interfaces/OCR2DRRegistryInterface.sol

```diff
@@ -0,0 +1,180 @@
+  }
+
+  /**
+   * @notice Get configuration relevant for making requests
+   * @return uint32 global max for request gas limit
+   * @return address[] list of registered DONs
+   */
+  function getRequestConfig() external view returns (uint32, address[] memory);
+
+  /**
+   * @notice Determine the charged fee that will be paid to the Registry owner
+   * @param data Encoded OCR2DR request data, use OCR2DRClient API to encode a request
+   * @param billing The request's billing configuration
+   * @return fee Cost in Juels (1e18) of LINK
+   */
+  function getRequiredFee(bytes calldata data, OCR2DRRegistryInterface.RequestBilling calldata billing)
+    external
+    view
+    returns (uint96);
+
+  /**
+   * @notice Estimate the execution cost in gas that will be reimbursed to the Node Operator who transmits the data on-chain
+   * @param billing The request's billing configuration
+   * @return gasAmount
+   */
+  function estimateExecutionGas(OCR2DRRegistryInterface.RequestBilling calldata billing)
+    external
+    view
+    returns (uint256);
+
+  /**
+   * @notice Estimate the total cost to make a request: gas re-imbursement, plus DON fee, plus Registry fee
+   * @param data Encoded OCR2DR request data, use OCR2DRClient API to encode a request
+   * @param billing The request's billing configuration
+   * @param donRequiredFee Fee charged by the DON that is paid to Oracle Node
+   * @return billedCost Cost in Juels (1e18) of LINK
+   */
+  function estimateCost(
+    bytes calldata data,
+    OCR2DRRegistryInterface.RequestBilling calldata billing,
+    uint96 donRequiredFee
+  ) external view returns (uint96);
+
+  /**
+   * @notice Initiate the billing process for an OCR2DR request
+   * @param data Encoded OCR2DR request data, use OCR2DRClient API to encode a request
+   * @param billing Billing configuration for the request
+   * @return requestId - A unique identifier of the request. Can be used to match a request to a response in fulfillRequest.
+   * @dev Only callable by an OCR2DROracle that has been approved on the Registry
+   */
+  function beginBilling(bytes calldata data, RequestBilling calldata billing) external returns (bytes32);
+
+  /**
+   * @notice Finalize billing process for an OCR2DR request
+   * @param requestId identifier for the request that was generated by the Registry in the beginBilling commitment
+   * @param response response data from DON consensus
+   * @param err error from DON consensus
+   * @param transmitter the Oracle who sent the report
+   * @param initialGas the initial amount of gas that was sent by the transmitter when submitting the report
+   * @return success whether the callback was successsful
+   * @return payment the amount charged to the subscription in Juels (1e18) of LINK
+   * @dev Only callable by OCR2DROracles that have been approved on the Registry
+   * @dev simulated offchain to determine if sufficient balance is present to fulfill the request
+   */
+  function concludeBilling(
```

Comment:
It would be nice if the name somehow indicated that user's callback is going to be called as part of it.

Reply:
`fulfillAndBill`? :grin:

Reply:
I support Bolek's point.

Reply:
billWithFullfillment? billWithCallback?